src/js/components/Calendar/Calendar.js

```diff
@@ -962,66 +973,73 @@ const Calendar = forwardRef(
-                if (!betweenDates(addDays(active, -7), displayBounds)) {
-                  changeReference(addDays(active, -7));
+                if (
+                  !!betweenDates(nextActive, normalizeInput(bounds)) ||
+                  !bounds
+                ) {
+                  setActive(nextActive);
+                  if (!betweenDates(nextActive, displayBounds)) {
+                    changeReference(nextActive);
+                  }
                 }
               }}
               onDown={(event) => {
+                const nextActive = addDays(active, 7);
                 event.preventDefault();
                 event.stopPropagation(); // so the page doesn't scroll
-                setActive(addDays(active, 7));
-                if (!betweenDates(addDays(active, 7), displayBounds)) {
-                  changeReference(active);
+                if (
+                  !!betweenDates(nextActive, normalizeInput(bounds)) ||
+                  !bounds
+                ) {
+                  setActive(nextActive);
+                  if (!betweenDates(nextActive, displayBounds)) {
+                    changeReference(nextActive);
+                  }
                 }
               }}
               onLeft={() => {
-                setActive(addDays(active, -1));
-                if (!betweenDates(addDays(active, -1), displayBounds)) {
-                  changeReference(active);
+                const nextActive = addDays(active, -1);
+                if (
+                  !!betweenDates(nextActive, normalizeInput(bounds)) ||
+                  !bounds
+                ) {
+                  setActive(nextActive);
+                  if (!betweenDates(nextActive, displayBounds)) {
+                    changeReference(nextActive);
+                  }
                 }
               }}
               onRight={() => {
-                setActive(addDays(active, 1));
-                if (!betweenDates(addDays(active, 2), displayBounds)) {
-                  changeReference(active);
+                const nextActive = addDays(active, 1);
+                if (
+                  !!betweenDates(nextActive, normalizeInput(bounds)) ||
+                  !bounds
+                ) {
+                  setActive(nextActive);
+                  if (!betweenDates(nextActive, displayBounds)) {
+                    changeReference(nextActive);
+                  }
                 }
               }}
             >
               <StyledWeeksContainer
-                tabIndex={0}
                 role="rowgroup"
                 aria-label={`${reference.toLocaleDateString(locale, {
                   month: 'long',
                   year: 'numeric',
                 })}; ${currentlySelectedString(value, locale)}`}
-                ref={daysRef}
                 sizeProp={size}
                 fillContainer={fill}
                 responsive={responsive}
-                focus={focus}
                 onFocus={() => {
```

Comment:
Are the onFocus/onBlur handlers needed anymore if this div won't have `tabIndex={0}`?

Reply:
So the `onFocus` is being called here more so as a function of the focus event bubbling up rather than the actual container receiving focus?